proposals/0088-enable-core-bpf-programs.md

```diff
@@ -0,0 +1,143 @@
+---
+simd: '0088'
+title: Enable Core BPF Programs
+authors:
+  - Joe Caulfield
+category: Standard
+type: Core
+status: Draft
+created: 2023-11-07
+feature: (fill in with feature tracking issues once accepted)
+---
+
+## Summary
+
+This proposal introduces the concept of Core BPF programs: programs which are
+essential to network operations. Currently, these are embedded programs known as
+"native" programs.
+
+This SIMD also outlines the processes by which engineers will manage upgrades to
+these enshrined Core BPF programs, as well as migrate existing native programs
+to core BPF.
+
+## Motivation
+
+With the advent of Program Runtime v2 and Firedancer, native programs will
+eventually become BPF programs.
+
+Program Runtime v2 will introduce a brand-new ABI for on-chain programs, and the
+native programs will need to be revised to support it.
+
+BPF programs offer less complexity for other clients, such as Firedancer, since
+developers will no longer have to keep up with program changes in their runtime
+implementations. Instead, the program can just be modified once. 
+
+For these reasons, it makes sense to introduce the concept of Core BPF programs:
+BPF programs the network depends on that should be upgraded with a special
+process.
+
+With an explicit process in place for upgrading Core BPF programs, engineers can
+safely manage changes to essential programs through feature gates. The feature
+gate process requries incremental activations on each cluster: testnet, then
+devnet, then mainnet-beta. This is extremely valuable for ensuring a change has
+been safely integrated and will not have negative effects on mainnet-beta.
+
+This same explicit process can be used to migrate existing native programs to
+their new Core BPF versions.
+
+## Alternatives Considered
+
+An alternative approach to managing upgrades to essential core programs could be
+to make these core BPF programs _upgradeable_ and establish a multi-signature
+upgrade authority. This authority could be comprised of key-holders from each
+validator client: Solana Labs, Jito, Jump, and possibly Syndica (Sig).
+
+However, this approach adds additional layers of complexity around upgrading
+these programs behind a feature-gate.
+
+## New Terminology
+
+- `Core BPF Program`: A **non-upgradeable** BPF program relied on by any part of
+  the Solana stack, including (but not limited to) consensus, transaction
+  processing, voting, staking, and account creation.
+- `Target Program`: The core program intending to be changed by some
+  feature activation.
+- `Source Program`: The modified version of a core program intending to be moved
+  in place of the existing core program.
```

Comment:
This doesn't seem like new terminology, and more about defining terms for the design. Let's move these into the detailed design portion